checkmedia.c

```diff
@@ -1,598 +1,205 @@
-  ) {
-    for(u = 0; u < ((iso.hybrid_size - iso.size) >> 1); u++) {
-      digest_media_process(&iso.digest.full_ctx, buffer, 2 << 10);
+    if(mediacheck_digest_valid(media->digest.part)) {
+      printf("   part ref: %s\n", mediacheck_digest_hex_ref(media->digest.part));
     }
   }
 
-  digest_media_finish(&iso.digest.ctx, iso.digest.current);
-  digest_media_finish(&iso.digest.full_ctx, iso.digest.full);
-
+  printf("   checking:     ");
+  fflush(stdout);
+  mediacheck_calculate_digest(media);
   printf("\n");
 
-  if(err) iso.err = 1;
-
-  iso.digest.got_current = 1;
-
-  if(iso.digest.got_old && !memcmp(iso.digest.current, iso.digest.old, iso.digest.size)) {
-    iso.digest.ok = 1;
-  }
-
-  close(fd);
-}
-
-
-/*
- * Read all kinds of iso header info.
- */
-void get_info(char *file, unsigned opt_verbose)
-{
-  int fd, ok = 0;
-  unsigned char buf[4];
-  char *s, *key, *value, *next;
-  unsigned u, u1, idx;
-  unsigned char mbr[512];
-
-  memset(&iso, 0, sizeof iso);
-
-  iso.err = 1;
-
-  if(!file) return;
-
-  if((fd = open(file, O_RDONLY | O_LARGEFILE)) == -1) return;
-
-  if(read(fd, mbr, sizeof mbr) == sizeof mbr) check_mbr(mbr);
-
-  if(
-    lseek(fd, 0x8028, SEEK_SET) == 0x8028 &&
-    read(fd, iso.vol_id, 32) == 32
-  ) {
-    iso.vol_id[sizeof iso.vol_id - 1] = 0;
-    ok++;
+  if(media->err && media->err_block) {
+    printf("        err: block %u\n", media->err_block);
   }
 
-  if(
-    lseek(fd, 0x8050, SEEK_SET) == 0x8050 &&
-    read(fd, buf, 4) == 4
-  ) {
-    iso.size = 2*(buf[0] + (buf[1] << 8) + (buf[2] << 16) + (buf[3] << 24));
-    ok++;
-  }
-
-  if(
-    lseek(fd, 0x823e, SEEK_SET) == 0x823e &&
-    read(fd, iso.app_id, 80) == 80
-  ) {
-    iso.app_id[sizeof iso.app_id - 1] = 0;
-    ok++;
-  }
-
-  if(
-    lseek(fd, 0x8373, SEEK_SET) == 0x8373 &&
-    read(fd, iso.app_data, 0x200) == 0x200
-  ) {
-    iso.app_data[sizeof iso.app_data - 1] = 0;
-    ok++;
-  }
-
-  close(fd);
-
-  if(ok != 4) return;
-
-  iso.media_type = "CD";
-
-  iso.err = 0;
-
-  for(s = iso.app_id + sizeof iso.app_id - 1; s >= iso.app_id; *s-- = 0) {
-    if(*s != 0 && *s != ' ') break;
-  }
-
-  if(!strncmp(iso.app_id, "MKISOFS", sizeof "MKISOFS" - 1)) *iso.app_id = 0;
-
-  if((s = strrchr(iso.app_id, '#'))) *s = 0;
-
-  if(strstr(iso.app_id, "DVD") || iso.size >= 1 << 20) iso.media_type = "DVD";
-
-  for(s = iso.vol_id + sizeof iso.vol_id - 1; s >= iso.vol_id; *s-- = 0) {
-    if(*s != 0 && *s != ' ') break;
+  printf("     result: ");
+  i = 0;
+  if(media->iso_blocks && mediacheck_digest_valid(media->digest.iso)) {
+    printf(
+      "iso %s %s",
+      mediacheck_digest_name(media->digest.iso),
+      mediacheck_digest_ok(media->digest.iso) ? "ok" : "wrong"
+    );
+    i = 1;
   }
-
-  if(*iso.vol_id == 'S') {
-    idx = iso.vol_id[strlen(iso.vol_id) - 1];
-    if(idx >= '1' && idx <= '9') iso.media_nr = idx - '0';
+  if(media->part_blocks && mediacheck_digest_valid(media->digest.part)) {
+    if(i) printf(", ");
```

Comment:
hmm, it is used to my taste a bit too away from initial i. Can it be named nicer like `iso` ? so here it will look like `if (iso) printf (", ")` which is more obvious why comma is needed in output?